File: backend/app/services/countries.py
from __future__ import annotations

COUNTRY_SEARCH_FOCUS = {
    "AE": (
        "gulf shipping",
        "Dubai logistics",
        "Abu Dhabi energy facilities",
        "air cargo disruption",
    ),
    "IN": (
        "heatwave",
        "power demand",
        "port congestion",
        "export logistics",
    ),
    "IR": (
        "Strait of Hormuz",
        "port blockade",
        "shipping restrictions",
        "energy exports",
    ),
    "KR": (
        "labor strike",
        "semiconductor production",
        "petrochemical supply",
        "export disruption",
    ),
    "OM": (
        "Strait of Hormuz",
        "Gulf of Oman",
        "shipping lanes",
        "maritime disruption",
    ),
    "SA": (
        "petrochemical complex",
        "energy infrastructure",
        "pipeline exports",
        "industrial disruption",
    ),
}

COUNTRY_NAMES = {
    "AE": "United Arab Emirates",
    "BR": "Brazil",
    "CA": "Canada",
    "CN": "China",
    "DE": "Germany",
    "FR": "France",
    "GB": "United Kingdom",
    "IN": "India",
    "IR": "Iran",
    "JP": "Japan",
    "KR": "South Korea",
    "MX": "Mexico",
    "OM": "Oman",
    "SA": "Saudi Arabia",
    "SG": "Singapore",
    "US": "United States",
    "VN": "Vietnam",
}


def country_name(country_code: str) -> str:
    normalized = country_code.strip().upper()
    return COUNTRY_NAMES.get(normalized, normalized)


def country_search_focus(country_code: str) -> tuple[str, ...]:
    normalized = country_code.strip().upper()
    return COUNTRY_SEARCH_FOCUS.get(normalized, ())


def map_watchlist_country_codes() -> list[str]:
    return sorted(COUNTRY_SEARCH_FOCUS)

File: frontend/lib/countries.ts
export const COUNTRY_NAME_BY_CODE: Record<string, string> = {
  AE: "United Arab Emirates",
  BR: "Brazil",
  CA: "Canada",
  CN: "China",
  DE: "Germany",
  FR: "France",
  GB: "United Kingdom",
  IN: "India",
  IR: "Iran",
  JP: "Japan",
  KR: "South Korea",
  MX: "Mexico",
  OM: "Oman",
  SA: "Saudi Arabia",
  SG: "Singapore",
  US: "United States",
  VN: "Vietnam"
};

export const COUNTRY_NUMERIC_BY_CODE: Record<string, string> = {
  AE: "784",
  BR: "076",
  CA: "124",
  CN: "156",
  DE: "276",
  FR: "250",
  GB: "826",
  IN: "356",
  IR: "364",
  JP: "392",
  KR: "410",
  MX: "484",
  OM: "512",
  SA: "682",
  US: "840",
  VN: "704"
};

export function countryNameFromCode(countryCode: string) {
  return COUNTRY_NAME_BY_CODE[countryCode] ?? countryCode;
}

export function countryNumericCode(countryCode: string) {
  return COUNTRY_NUMERIC_BY_CODE[countryCode];
}
